tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/repository/CompositeDataTypeMappingTransformationTest.xtend

```diff
@@ -0,0 +1,134 @@
+
+import static extension edu.kit.ipd.sdq.commons.util.java.lang.IterableUtil.*
+import tools.vitruv.applications.pcmjava.tests.pcm2java.javahelper.FluentJavaClassBuilder
+
+class CompositeDataTypeMappingTransformationTest extends Pcm2JavaTransformationTest {
+	
+	@Test
+	def void testAddCompositeDataType() {
+		createRepository(Pcm2JavaTestUtils.REPOSITORY_NAME)
+		
+		changePcmView[
+			modifySingleRepository[
+				dataTypes__Repository += createCompositeDataType(Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME)
+			]
+		]
+		
+		validateJavaView[
+			val expectedCompilationUnit = new FluentJavaClassBuilder(
+				Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME,
+				Pcm2JavaTestUtils.REPOSITORY_NAME + Pcm2JavaTestUtils.DATATYPES_SUFIX
+			).build
+			
+			assertExistenceOfCompilationUnitsDeeplyEqualTo(List.of(expectedCompilationUnit))
+		]
+	}
+
+	@Test
+	def void testRenameCompositeDataType() {
+		createRepository(Pcm2JavaTestUtils.REPOSITORY_NAME)
+		changePcmView[
+			modifySingleRepository[
+				dataTypes__Repository += createCompositeDataType(Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME)
+			]
+		]
+		
+		changePcmView[
+			modifySingleRepository[
+				var dataType = dataTypes__Repository.filter(CompositeDataType).claimOne
+				dataType.entityName = Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME + Pcm2JavaTestUtils.RENAME
+			]
+		]
+		
+		validateJavaView[
+			val expectedCompilationUnit = new FluentJavaClassBuilder(
+				Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME + Pcm2JavaTestUtils.RENAME,
+				Pcm2JavaTestUtils.REPOSITORY_NAME + Pcm2JavaTestUtils.DATATYPES_SUFIX
+			).build
+			
+			assertExistenceOfCompilationUnitsDeeplyEqualTo(List.of(expectedCompilationUnit))
+		]
+	}
+
+	@Test
+	def void testAddPrimitiveTypeToCompositeDataType() {
+		createRepository(Pcm2JavaTestUtils.REPOSITORY_NAME)
+		changePcmView[
+			modifySingleRepository[
+				dataTypes__Repository += createCompositeDataType(Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME)
+				dataTypes__Repository += createPrimitiveDataType(PrimitiveTypeEnum.INT)
+			]
+		]
+		
+		changePcmView[
+			modifySingleRepository[
+				val innerType = dataTypes__Repository.filter(PrimitiveDataType).claimOne
+				val dataType = dataTypes__Repository.filter(CompositeDataType).claimOne
+				dataType.innerDeclaration_CompositeDataType += createInnerDeclaration(innerType, Pcm2JavaTestUtils.INNER_DEC_NAME)
+			]
+		]
+		
+		validateJavaView[
+			val expectedCompilationUnit = new FluentJavaClassBuilder(
+					Pcm2JavaTestUtils.COMPOSITE_DATA_TYPE_NAME, 
+					Pcm2JavaTestUtils.REPOSITORY_NAME + Pcm2JavaTestUtils.DATATYPES_SUFIX
+				)
+				.addPrivateField(Pcm2JavaTestUtils.INNER_DEC_NAME, createInt())
+				.addGetterForField(Pcm2JavaTestUtils.INNER_DEC_NAME)
+				.addSetterForField(Pcm2JavaTestUtils.INNER_DEC_NAME)
+				.build
+			
+			assertExistenceOfCompilationUnitsDeeplyEqualTo(List.of(expectedCompilationUnit))
+		]
+	}
+	
+	@Test
+	def void testAddCompositeDataTypeToCompositeDataType() {
```

Comment:
all strings in this method to constants (can be constants in the scope of the test class instead of in the utility class)